src/pages/MonthlyGood/MonthlyGoodThanksPage.vue

```diff
@@ -81,34 +71,39 @@ export default {
 			dayOfMonth: new Date().getDate(),
 			// user flags
 			isMonthlyGoodSubscriber: false,
-			isOnetimePayment: null,
 			autoDepositId: null,
 			category: null,
 			mdiCheckCircle,
 		};
 	},
 	inject: ['apollo', 'cookieStore'],
 	apollo: {
-		query: pageQuery,
-		preFetch: true,
-		result({ data }) {
-			this.isMonthlyGoodSubscriber = _get(data, 'my.autoDeposit.isSubscriber', false);
-			if (!this.isMonthlyGoodSubscriber) {
-				this.$router.push({ path: '/monthlygood' });
-			}
-			this.autoDepositAmount = numeral(_get(data, 'my.autoDeposit.amount', 0)).format('0.00');
-			this.donation = numeral(_get(data, 'my.autoDeposit.donateAmount', 0)).format('0.00');
-			this.mgAmount = this.autoDepositAmount - this.donation;
-			this.dayOfMonth = _get(data, 'my.autoDeposit.dayOfMonth');
-			this.category = _get(data, 'my.monthlyGoodCategory');
-			this.isOnetimePayment = _get(data, 'my.autoDeposit.isOnetime');
-			this.autoDepositId = _get(data, 'my.autoDeposit.id');
+		preFetch(config, client) {
+			return client.query({
+				query: pageQuery
+			}).then(({ data }) => {
+				this.isMonthlyGoodSubscriber = _get(data, 'my.autoDeposit.isSubscriber', false);
+
+				if (!this.isMonthlyGoodSubscriber) {
+					// Return user to /monthlygood
+					return Promise.reject({ path: '/monthlygood' });
+				}
+
+				this.autoDepositAmount = numeral(_get(data, 'my.autoDeposit.amount', 0)).format('0.00');
+				this.donation = numeral(_get(data, 'my.autoDeposit.donateAmount', 0)).format('0.00');
+				this.mgAmount = this.autoDepositAmount - this.donation;
+				this.dayOfMonth = _get(data, 'my.autoDeposit.dayOfMonth');
+				this.category = _get(data, 'my.monthlyGoodCategory');
+				this.autoDepositId = _get(data, 'my.autoDeposit.id');
```

Comment:
Generally, `this` isn't available in the then block of the preFetch function. You'll need to either keep the `result` block below or read the query in created and then set your data vars... ref. ContentfulPage, AutolendSettingsPage + CheckoutPage vue file.

Reply:
Oh right, good call. I was testing the data being returned but not being set.

Reply:
I went back and did some testing, the`this.` settings worked (what I saw), but `this` here is the `preFetch` function itself! That's not useful... Haha. Anyways, I backtracked to the original error and realized that the error could be resolved with just adding a `.catch` to the router push, since the push still goes through if you quiet the unimportant error it was throwing.